src/schoology-extractor/tests/mapping/test_users.py

```diff
@@ -0,0 +1,86 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+
+import pandas as pd
+import pytest
+
+from schoology_extractor.mapping.users import map_to_udm
+
+
+class Test_mapping_schoology_users_to_udm:
+    @classmethod
+    def setup_class(cls):
+
+        # Arrange
+        users_csv = """uid,id,school_id,synced,school_uid,name_title,name_title_show,name_first,name_first_preferred,use_preferred_first_name,name_middle,name_middle_show,name_last,name_display,username,primary_email,picture_url,gender,position,grad_year,password,role_id,tz_offset,tz_name,parents,child_uids,language,additional_buildings
+100032890,100032890,2908525646,0,604863,,0,Mary,,1,Catherine,0,Archer,Mary Archer Display Name,mary.archer,mary.archer@studentgps.org,https://asset-cdn.schoology.com/system/files/imagecache/profile_reg/sites/all/themes/schoology_theme/images/user-default.gif,,,,,123456,-5,America/Chicago,,,,
+99785799,99785799,2908525646,0,222222,,0,Brad,,1,,0,Banister,Brad Banister,brad.banister,brad@doublelinepartners.com,https://asset-cdn.schoology.com/system/files/imagecache/profile_reg/sites/all/themes/schoology_theme/images/user-default.gif,,,,,123457,-5,America/Chicago,,,,"""
+        lines = users_csv.split("\n")
+        users_df = pd.DataFrame(
+            [x.split(",") for x in lines[1:]], columns=lines[0].split(",")
+        )
+
+        roles_csv = """id,title,faculty,role_type
+123456,student,0,1
+123457,teacher,0,1"""
+        lines = roles_csv.split("\n")
+        roles_df = pd.DataFrame(
+            [x.split(",") for x in lines[1:]], columns=lines[0].split(",")
+        )
+
+        # Act
+        cls.result = map_to_udm(users_df, roles_df)
+
+    # Each assertion is a separate method
+    def test_then_output_has_two_rows(self):
+        assert self.result.shape[0] == 2
+
+    def test_then_output_has_ten_columns(self):
+        assert self.result.shape[1] == 10
+
+    @pytest.mark.parametrize(
+        "input",
+        [
+            "SourceSystemIdentifier",
+            "SourceSystem",
+            "UserRole",
+            "LocalUserIdentifier",
+            "SISUserIdentifier",
+            "Name",
+            "EmailAddress",
+            "EntityStatus",
+            "CreateDate",
+            "LastModifiedDate",
+        ],
+    )
+    def test_then_output_has_column(self, input):
+        assert input in self.result.columns

```

Comment:
@bradbanister I'm interested in seeing and possibly using `describe`. For now, however, I was able to get the results I wanted simply by using the `setup_class` method.